zio/detector/reader.go

```diff
@@ -59,6 +60,52 @@ func NewReader(r io.Reader, zctx *resolver.Context) (zbuf.Reader, error) {
 	return nil, joinErrs([]error{tzngErr, zeekErr, ndjsonErr, zjsonErr, zngErr})
 }
 
+func Format(r io.Reader) (string, error) {
```

Comment:
I don't see why you've duplicated all of the detection logic. This is only called from logOp.openFile; there should be a way for it to call detector.OpenFile.

Reply:
This is the essence of detector should be about. Aside from this its a little complicated why I did this: In log ingest endopoint, if a file was getting rejected by detector.NewReader, the bytes read would get counted towards the total of bytes read in the log ingest op. To me the simplest solution was to use this function to detetect the stream type then set the underlying file seek to `SeekStart` and pass the reset file into `LookupReader`.

Reply:
Dropped it and went back to readCounter.